feature-libs/cart/main/components/cart/added-to-cart-dialog/added-to-cart-dialog.component.ts

```diff
@@ -82,24 +81,39 @@ export class AddedToCartDialogComponent implements OnInit {
     return this.quantityControl$;
   }
 
-  ngOnInit() {
-    this.addedEntryWasMerged$ = this.loaded$.pipe(
-      filter((loaded) => loaded),
-      switchMapTo(this.cartService.getEntries()),
-      map((entries) => entries.length === this.numberOfEntriesBeforeAdd)
+  init(
+    productCode: string,
+    quantity: number,
+    numberOfEntriesBeforeAdd: number
+  ): void {
+    // Display last entry for new product code. This always corresponds to
+    // our new item, independently of whether merging occured or not
+    this.entry$ = this.activeCartFacade.getLastEntry(productCode);
+    this.quantity = quantity;
+    this.addedEntryWasMerged$ = this.getAddedEntryWasMerged(
+      numberOfEntriesBeforeAdd
     );
   }
 
+  protected getAddedEntryWasMerged(
+    numberOfEntriesBeforeAdd: number
+  ): Observable<boolean> {
+    return this.loaded$.pipe(
+      filter((loaded) => loaded),
+      switchMapTo(this.activeCartFacade.getEntries()),
+      map((entries) => entries.length === numberOfEntriesBeforeAdd)
+    );
+  }
   /**
    * Adds quantity and entryNumber form controls to the FormGroup.
    * Returns quantity form control.
    */
-  protected getQuantityFormControl(entry: OrderEntry): FormControl {
+  protected getQuantityFormControl(entry: OrderEntry | undefined): FormControl {
```

Comment:
can be this `protected getQuantityFormControl(entry?: OrderEntry): FormControl`

Reply:
`(entry?: OrderEntry)` is the same as `(entry: OrderEntry | undefined)`, right? What alternative you want? I am also fine with this `protected getQuantityFormControl(entry: OrderEntry | undefined)`. :)

Reply:
I'm sorry, my bad.  I misread your first comment.  And I meant that I was going to try to find alternatives to what I did.

I applied your suggestion : )